tasks/drupal/deploy.yml

```diff
@@ -0,0 +1,164 @@
+    checksum: "{{ steamengine_build_checksum }}"
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: u=rwx,g=rx,o=
+    headers: "{{ steamengine_build_url_headers }}"
+    validate_certs: "{{ steamengine_build_url_validate_certs }}"
+  notify:
+    - "{{ steamengine_project_name }} fixpermissions"
+    - "{{ steamengine_project_name }} restart"
+  register: get_url_build
+  tags:
+    - steamengine_deploy_drupal
+
+- name: "Is there a new build: {{ get_url_build.changed }}"
+  set_fact:
+    new_build_to_deploy: "{{ get_url_build.changed|bool }}"
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Create tmp directory
+  tempfile:
+    state: directory
+    suffix: "{{ steamengine_project_name }}"
+  register: tempdir
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_sf
+
+- name: Unarchive build
+  unarchive:
+    src: "{{ steamengine_project_root_path }}/project.zip"
+    dest: "{{ tempdir.path }}"
+    remote_src: true
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_sf
+
+- name: Ensure settings
+  copy:
+    src: "{{ tempdir.path }}/web/sites/default/default.settings.php"
+    dest: "{{ steamengine_conf_path }}/settings.php"
+    force: no
+    remote_src: true
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: u=rwx,g=rx,o=
+  register: drupal_settings
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Customize drupal settings
+  blockinfile:
+    dest: "{{ steamengine_conf_path }}/settings.php"
+    block: "{{ steamengine_drupal_project_setting_block }}"
+    mode: u=rwx,g=rx,o=
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Clean www directory
+  file:
+    path: "{{ steamengine_project_root_path_web }}"
+    state: absent
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: "Move archive files to {{ steamengine_project_root_path_web }}"
+  command: "mv {{ tempdir.path }} {{ steamengine_project_root_path_web }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Remove tmp directory
+  file:
+    path: "{{ tempdir.path }}"
+    state: absent
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Create configuration link
+  file:
+    state: link
+    src: "{{ steamengine_conf_path }}/settings.php"
+    dest: "{{ steamengine_project_root_path_web }}/web/sites/default/settings.php"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Create link directory
+  file:
+    state: link
+    src: "{{ steamengine_drupal_persistent_path }}/{{ item.key }}"
+    dest: "{{ steamengine_project_root_path_web }}/{{ item.value }}"
```

Comment:
Sometimes the dest directory is present in the build and the deployment crash:

```
failed: [xx-website-preprod-1] (item={'key': 'private_files', 'value': 'drupal_private_files'}) => changed=false 
  ansible_loop_var: item
  gid: 1004
  group: xx-websiteapp
  item:
    key: private_files
    value: drupal_private_files
  mode: '0777'
  msg: the directory /xx-website/project_root/www/drupal_private_files is not empty, refusing to convert it
  owner: xx-website
  path: /xx-website/project_root/www/drupal_private_files
  size: 4096
  state: directory
  uid: 1004
```
We could add a step to delete the dest directory so that we don't have this issue.